cai_causal_graph/graph_components.py

```diff
@@ -310,6 +326,46 @@ def __hash__(self) -> int:
         """Return a hash value of the node identifier."""
         return super().__hash__()
 
+    def to_dict(self, include_meta: bool = True) -> dict:
+        """
+        Serialize a `cai_causal_graph.graph_components.TimeSeriesNode` instance to a dictionary.
+
+        :param include_meta: Whether to include meta information about the node in the dictionary. Default is `True`.
+        :return: The dictionary representation of the `cai_causal_graph.graph_components.TimeSeriesNode` instance.
+        """
+        dictionary = super().to_dict(include_meta)
+        # add the time lag and variable name to the dictionary
+        dictionary.update({TIME_LAG: self.time_lag, VARIABLE_NAME: self.variable_name})
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> TimeSeriesNode:
+        """
+        Return a `cai_causal_graph.graph_components.TimeSeriesNode` instance from a dictionary.
+
+        :param dictionary: The dictionary from which to create the time series node.
+        :return: The time series node created from the dictionary.
+        """
+        assert 'identifier' in dictionary, 'The dictionary must contain the `identifier` key.'
+
+        if TIME_LAG not in dictionary or VARIABLE_NAME not in dictionary:
+            dictionary = dictionary.copy()
+            # reconstruct from identifier
+            variable_name, time_lag = get_variable_name_and_lag(dictionary.get('identifier'))  # type: ignore
+            dictionary[TIME_LAG] = time_lag
+            dictionary[VARIABLE_NAME] = variable_name
+
+        return cls(
+            identifier=dictionary.get('identifier'),
+            time_lag=dictionary.get(TIME_LAG),
+            variable_name=dictionary.get(VARIABLE_NAME),
+            meta=dictionary.get('meta', None),
+            variable_type=dictionary.get('variable_type', NodeVariableType.UNSPECIFIED),
+        )
+
+
+NodeClassDict = {'Node': Node, 'TimeSeriesNode': TimeSeriesNode}
```

Comment:
Would put at top of the file.

Reply:
It does not work at the top.

Reply:
Because the interpreter cannot resolve the symbols

Reply:
Ah yes. That's obvious now. Duh.